django/db/models/fields/tuple_lookups.py

```diff
@@ -28,17 +28,31 @@ def __iter__(self):
 
 class TupleLookupMixin:
     def get_prep_lookup(self):
+        self.check_rhs_is_tuple_or_list()
         self.check_rhs_length_equals_lhs_length()
         return self.rhs
 
+    def check_rhs_is_tuple_or_list(self):
+        if not isinstance(self.rhs, (tuple, list)):
+            lhs_str = self.get_lhs_str()
+            raise ValueError(
+                f"'{self.lookup_name}' lookup of {lhs_str} must be a tuple or a list"
+            )
+
     def check_rhs_length_equals_lhs_length(self):
         len_lhs = len(self.lhs)
         if len_lhs != len(self.rhs):
+            lhs_str = self.get_lhs_str()
             raise ValueError(
-                f"'{self.lookup_name}' lookup of '{self.lhs.field.name}' field "
-                f"must have {len_lhs} elements"
+                f"'{self.lookup_name}' lookup of {lhs_str} must have {len_lhs} elements"
             )
 
+    def get_lhs_str(self):
+        if isinstance(self.lhs, ColPairs):
+            return f"'{self.lhs.field.name}'"
```

Comment:
I think it's slightly cleaner to use `repr`:

```suggestion
            return repr(self.lhs.field.name)
```